### ISSUE TITLE
replace SqlServerSchemaReader  to DatabaseSchemaReader,then will support all database

### ISSUE THREAD

**Issue by liminany** [ReversePOCO/EntityFramework-Reverse-POCO-Code-First-Generator#26]
The first,It's a nice work .But I work for mysql/Oracle,so hope support other database,You can check the Lib [DatabaseSchemaReader](https://github.com/martinjw/dbschemareader),Thanks!

**Comment by luylucas10** [comment]
Yes, it's a great tool and I would like to use it with postgres.

**Comment by CodingSamurai** [comment]
Another vote for Oracle (if necessary via ODP.net) support.

**Comment by mahosi1** [comment]
Any update on this?

**Comment by sjh37** [comment]
Sorry, no update. Work keeps getting in the way of important progress on this project :-)
I was going to look at the reverse engineering code within EntityFramework itself, although I suspect it won't pull in everything I need (but it might...)

**Comment by sjh37** [comment]
Done. v4 supports SQL Server, PostgreSQL, MySQL, Oracle and SQLite, with the database reading done by the `efrpg` dotnet tool.
